### PR TITLE
Feat/new search

### PR DESCRIPTION
**Pull request by jennieramida**

### CODE REVIEW

**Comment by tansawit** [comment]
small bug. most likely API level?

CleanShot 2567-07-04 at 12 28 20@2x

File: src/lib/layout/SearchResultRenderer.tsx
Comment (Poafs1) [inline src/lib/layout/SearchResultRenderer.tsx:98]: Can you move this to interface pls.

File: src/lib/layout/SearchResultRenderer.tsx
Comment (Poafs1) [inline src/lib/layout/SearchResultRenderer.tsx:278]: ditto

**Review by Poafs1 — APPROVED** [review]
lgtm. pls wait for Nat and Peach to review before merge

File: src/lib/layout/search/SearchComponent.tsx
Comment (evilpeach) [inline src/lib/layout/search/SearchComponent.tsx:191]: Change to this

```ts
if (isMobile) return (<Button variant="outline-gray" size="sm" onClick={onOpen}>
          <CustomIcon name="search" boxSize={3} />
        </Button>)

return (<Flex>...</Flex>)
```

File: src/lib/layout/search/SearchZeroState.tsx
Comment (evilpeach) [inline src/lib/layout/search/SearchZeroState.tsx:18]: starter => base

File: src/lib/layout/search/SearchZeroState.tsx
Comment (evilpeach) [inline src/lib/layout/search/SearchZeroState.tsx:43]: Please add key.

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/layout/search/SearchComponent.tsx
Comment (jennieramida) [reply src/lib/layout/search/SearchComponent.tsx:191]: can't, use same modal below

File: src/lib/layout/search/SearchDisplayResult.tsx
Comment (songwongtp) [inline src/lib/layout/search/SearchDisplayResult.tsx:16]: SearchResultItemBody

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by Poafs1 — APPROVED** [review]

**Pull request merged**
